components/get_buddies.php

```diff
@@ -13,29 +13,28 @@
     // Retrieve accepted buddies for the current user
     $user_id = $_SESSION['user_id'];
 
-    $fetch_buddies_sql = "SELECT * FROM buddies WHERE (user1_id = :user_id OR user2_id = :user_id) AND status = 'accepted'";
+    $fetch_buddies_sql = "
+        SELECT u.username, u.email
+        FROM buddies b
+        JOIN users u ON b.user1_id = u.id
+        WHERE (b.user2_id = :user_id)
+        AND b.status = 'accepted'";
     $fetch_buddies_stmt = $pdo->prepare($fetch_buddies_sql);
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (bug_risk):** Query change may exclude some buddies.

The revised SQL query only fetches buddies where the current user is user2. Ensure this change doesn't unintentionally exclude buddies where the current user is user1.